sdk/python/feast/infra/local.py

```diff
@@ -60,9 +61,7 @@ def update_infra(
         for table in tables_to_delete:
             conn.execute(f"DROP TABLE IF EXISTS {_table_id(project, table)}")
 
-    def teardown_infra(
-        self, project: str, tables: Sequence[Union[FeatureTable, FeatureView]]
-    ) -> None:
+    def teardown_infra(self, *args, **kwargs,) -> None:
```

Comment:
Is there a reason to not have consistent arguments across methods for implementations of `Provider`?

Reply:
Nor all arguments are necessary as it depends on the Provider itself (as here we don't use any of the Parameters).
For example on the "ShopifyProvider", we would not use the `project` parameter neither.

Therefore I thought that only using the arguments we need is a good practice and it makes the upstream Provider object more flexible (if we want to add other parameters in the future)

Happy to hard code everything. Don't have strong feelings about it

Reply:
One part that gives me pause: We'd lose our type checking if we go this route. If I (personally) were to implement a `Provider` I would want to know when the upstream project (`Feast`) has made a change to parameters that will be passed to me, and ideally not at runtime.

Reply:
Fair enough 👍 

Will go that route then

Reply:
Done